protocols/imap/src/main/java/org/apache/james/imap/processor/SearchProcessor.java

```diff
@@ -265,10 +265,6 @@ private ModSeq findHighestModSeq(MailboxSession session, MessageManager mailbox,
 
     private SearchQuery toQuery(SearchKey key, ImapSession session) throws MessageRangeException {
         final SearchQuery result = new SearchQuery();
-        final SelectedMailbox selected = session.getSelected();
-        if (selected != null) {
-            result.addRecentMessageUids(selected.getRecent());
```

Comment:
there is no more way to search for recent emails, or did I miss something?

Reply:
There is: by querying directly what's stored.